src/python_testing/matter_testing_support.py

```diff
@@ -365,6 +401,9 @@ class ProblemNotice:
     problem: str
     spec_location: str = ""
 
+    def __str__(self):
+        return f'\nProblem: {str(self.severity)}\n    test_name: {self.test_name}\n    location: {str(self.location)}\n    problem: {self.problem}\n    spec_location: {self.spec_location}\n'
```

Comment:
nit: Maybe it will be better to use "multiline" formating like in other __str__?